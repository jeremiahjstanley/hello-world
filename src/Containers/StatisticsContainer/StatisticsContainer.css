h1, a {
  font-family: Avenir Next;
  font-weight: 100;
}

h1 {
  font-size: 2em;
}

a {
  font-size: 1.25em;
  font-weight: 300;
  text-decoration: none;
  color: #000000;
}

.statistics-container {
	width: 80%;
	margin: auto;
}