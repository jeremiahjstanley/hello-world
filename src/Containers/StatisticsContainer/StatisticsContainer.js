import React from 'react';
import { connect } from 'react-redux';
import GovernanceIndicatorsStatistics from '../GovernanceIndicatorsStatistics/GovernanceIndicatorsStatistics';
import './StatisticsContainer.css'


export const  StatisticsContainer = ({dataBase}) => {
  return (
    <div>
      <GovernanceIndicatorsStatistics/>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  dataBase: state.dataBase,
});

// create switch statement for WWDI figures

export default connect(mapStateToProps)(StatisticsContainer);