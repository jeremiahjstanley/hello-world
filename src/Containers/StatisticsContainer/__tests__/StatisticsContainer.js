import React from 'react';
import { shallow } from 'enzyme';
import { StatisticsContainer, mapStateToProps } from '../StatisticsContainer'

describe('StatisticsContainer', () => {
  let wrapper;
  let mockInitialLocation;

  beforeEach(() => {
    mockInitialLocation = []
    wrapper = shallow(
      <StatisticsContainer
        initialLocation={mockInitialLocation}
      />)
  })
  describe('StatisticsContainer unit tests', () => {
    it('should match the snapshot', () => {

      expect(wrapper).toMatchSnapshot();
    })
  })
  describe('mapStateToProps', () => {
    it('should return a props object with an intial location array', () => {
      const mockState = {initialLocation: [{ name: '2016-12-31', Estimate: 1.3778049945831, 'Number of Sources': 11,'Percentile Rank': 96.059112548828, 'Standard Error': 0.13451327383518 }]};
      const expected = {initialLocation: mockState.initialLocation}
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected);
    })
  })
})