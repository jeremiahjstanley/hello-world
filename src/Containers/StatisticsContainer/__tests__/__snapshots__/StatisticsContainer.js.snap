// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsContainer StatisticsContainer unit tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dataBase={
        Object {
            "database_code": "WWGI",
            "name": "Worldwide Governance Indicators",
          }
    }
    dataSet={
        Object {
            "dataset_code": "VA",
            "name": "Voice and Accountability",
          }
    }
    location={
        Object {
            "alpha_2": "BE",
            "alpha_3": "BEL",
            "country_code": "056",
            "intermediate_region": "",
            "intermediate_region_code": "",
            "iso_3166_2": "ISO 3166-2: BE",
            "name": "Belgium",
            "region": "Europe",
            "region_code": "150",
            "sub_region": "Western Europe",
            "sub_region_code": "155",
          }
    }
    locationData={
        Array [
            Object {
              "Estimate": 1.3778049945831,
              "Number of Sources": 11,
              "Percentile Rank": 96.059112548828,
              "Standard Error": 0.13451327383518,
              "name": "2016-12-31",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Worldwide Governance Indicators
           - 
          Voice and Accountability
</h1>,
        <Connect(GovernanceIndicatorsStatistics) />,
        <withRouter(Connect(Component)) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Worldwide Governance Indicators",
            " - ",
            "Voice and Accountability",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Worldwide Governance Indicators",
          " - ",
          "Voice and Accountability",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Worldwide Governance Indicators
             - 
            Voice and Accountability
</h1>,
          <Connect(GovernanceIndicatorsStatistics) />,
          <withRouter(Connect(Component)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Worldwide Governance Indicators",
              " - ",
              "Voice and Accountability",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Worldwide Governance Indicators",
            " - ",
            "Voice and Accountability",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
